AUT-575, AUT-576, AUT-577, AUT-582

mmattano opened the pull request:
Transferred from personal repository: .py file with relevant functions, an example case with simplified notebook, tests and test data, a short description in the docs, addition to the requirements

mmattano committed:
feat: Added INCA MFA tools

Transferred from personal repository: .py file with relevant functions, an example case with simplified notebook, tests and test data, a short description in the docs, addition to the requirements

mmattano committed:
fix: adjusted names to import functions

mmattano committed:
fix: adjusted using black and quick fix for line length issues

Added "# noqa E501" to lines that were to long where I could not just break them (the strings that are necessary for setting up the MATLAB file) because breaking them caused issues with the resulting script

mmattano committed:
fix: forgot one file

mmattano commented:
Ready to be added, additional features will be built right away in this repository

svevol commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1050 @@
+
+def limit_to_one_model(data_input, model_name_column, model_name):
+    """
+    Limits the data to values that are assigned to one metabolic model
```
Can you add the argument and return sections to the docstrings?

svevol commented on requirements.txt:
```diff
@@ -1,2 +1,5 @@
-pytfa
\ No newline at end of file
+pytfa
+#INCA_tools
```
I think pytfa installs numpy and pandas. Since there is going to be more requirements in this repo I propose to specify the version of pandas and numpy if you think that it is necessary. Cobra sometimes has incompatibilities with wrong version of numpy or pandas.

mmattano committed:
fix: Adjusted according to comments

"Can you add the argument and return sections to the docstrings?"

mmattano committed:
fix: adjustment of requirements.txt according to comments

"I think pytfa installs numpy and pandas. Since there is going to be more requirements in this repo I propose to specify the version of pandas and numpy if you think that it is necessary. Cobra sometimes has incompatibilities with wrong version of numpy or pandas."

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+    data_input, experiment_name_column, experiment_name
+):
+    """
+    Limits the data to values that were aquired in one experiment
```
typo with "aquired"

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+                    duplicates_not_done = False
+
+                # here we check if it is a tuple, i.e. there is more than one
+                # metabolite, or a a list, i.e. only one metabolite
```
please remove the duplicate "a" in the comment.

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+                    dupes_molecule.append(x)
+                seen_molecule[x] += 1
+
+    # Iterate through the metabolites involved in the reactions
```
Thanks for adding in these detailed comments.  This helps a lot in better understanding the code since the data structures are a bit hard to understand without actually running the code.

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+    """
+    mat_script = "r = reaction({... % define reactions\n"
+
+    # this is the teporary biomass function
```
Small type with "teporary"

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+
+    # this is the teporary biomass function
+    # biomass_INCA_iJS2012
+    a = "0.176*phe_DASH_L_c + 0.443*mlthf_c + 0.34*oaa_c + 0.326*lys_DASH_L_c + 33.247*atp_c + 0.205*ser_DASH_L_c + 0.129*g3p_c + 0.131*tyr_DASH_L_c + 0.051*pep_c + 0.146*met_DASH_L_c + 0.205*g6p_c + 0.087*akg_c + 0.25*glu_DASH_L_c + 0.25*gln_DASH_L_c + 0.754*r5p_c + 0.071*f6p_c + 0.083*pyr_c + 0.582*gly_c + 0.241*thr_DASH_L_c + 0.229*asp_DASH_L_c + 5.363*nadph_c + 0.087*cys_DASH_L_c + 0.619*3pg_ ...
```
I recommend we do something that is a lot more extensible than the hard coding of the biomass function as I did in my original scripts.  

two thoughts come to mind:
1. We could add it in as a parameter to the method.
2. We could add it to the .csv file so that it is added in automatically (given this is possible)

mmattano replied:
I would add it as a parameter for now and make sure to include a catalogue for different organisms once I add it later on for actual users

mmattano replied:
Added the biomass function to the example notebook for now

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+                        for reactant_cnt, reactant in enumerate(
+                            model_reactants_ids
+                        ):
+                            # this little bit adds +s if it's not
```
Were you having issues with duplicate reactions?  There should not be duplicate reactions (nor duplicate metabolites).  If duplicates are found, I would recommend aborting and alerting the user that something is wrong with the input.

mmattano replied:
One reaction was duplicated which cause quite an issue, I had to remove it manually. I can add something that checks for that and alerts the user

mmattano replied:
And the duplicate metabolites in that step are fine, that happens if one metabolite that can have more than one mapping is part of the reaction

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+        x["met_id"] for cnt, x in atomMappingMetabolite_data_I.iterrows()
+    ]
+    # unbalanced reactions:
+    for met in ["co2_e", "h2o_e", "h_e", "na1_e"]:
```
I would recommend moving this list to a default parameter so the user has a bit more control over the "hard coding"

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+    model_rxn_ids,
+    fluxes_present=True,
+):
+    # Stedv stuff should be checked once more
```
I think this comment will break the docstrings rendering?

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+
+
+def unbalanced_reactions(atomMappingMetabolite_data_I):
+    # NOTE: hard-coded for now until a better workaround can be done
```
I think this comment will break the docstrings rendering? I would suggest to remove or add to the docstrings itself.

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -0,0 +1,1232 @@
+
+
+def mapping(experimentalMS_data_I, fragments_used):
+    # check stedv stuff
```
I think this comment will break the docstrings rendering? I would suggest to remove or add to the docstrings itself.

dmccloskey commented on tests/test_INCA_script_generator.py:
```diff
@@ -0,0 +1,56 @@
+import sys
+import pathlib
+
+sys.path.insert(1, "../")
```
You did an amazing job of refactoring the code into discrete methods.  I would recommend adding in unit tests for each of the discrete functions so that if something goes wrong, we know exactly which of the refactored functions is the cause.

dmccloskey reviewed (changes requested):
Wow! What a huge improvement over what I originally wrote!  Very nice work.

I have several small comments involving typos and docstrings.  The major issue I see is unit test coverage for all of the smaller functions.  I would recommend setting those in place before we merge.

mmattano committed:
fix: corrected typos

Typos that were pointed in PR comments have been adjusted

mmattano committed:
feat: biomass function is now an editable parameter

mmattano committed:
fix: moved comments to not mess with docstrings

mmattano committed:
feat: the unbalanced metabolites are now a parameter

mmattano committed:
fix: changes script for runner

mmattano committed:
fix: changed because of addition of biomass function

mmattano committed:
feat: added biomass function as a parameter

mmattano committed:
fix: adapted and generated test data

mmattano committed:
Merge branch 'develop' into MM_INCA_tools

mmattano committed:
fix: added error message if there is a duplicate reaction

mmattano committed:
Merge branch 'MM_INCA_tools' of https://github.com/mmattano/AutoFlow-OmicsDataHandling into MM_INCA_tools

mmattano committed:
Update INCA_script_generator.py

mmattano committed:
feat: changed test data generation and added the new obj

mmattano committed:
tests: added new tests for most of the custom functions

mmattano committed:
fix: black caused an issue with the data import

mmattano committed:
tests: small adjustments and comments

dmccloskey commented on AutoFlow_OmicsDataHandling/INCA_script_generator.py:
```diff
@@ -273,11 +276,17 @@ def add_reactions_to_script(modelReaction_data_I, atomMappingReactions_data_I):
         model_rxn_ids_exp: List of reaction IDs used for the model
 
     """
+    if len(atomMappingReactions_data_I["rxn_id"]) != len(
```
If you don't mind adding in the new `biomass_function` parameter to the docstrings that would be great.

dmccloskey replied:
Perhaps it would also be a good idea to start a separate module with different default biomass functions for the user to choose if it is problematic reading from a .csv file.

mmattano replied:
Hey so I changed it a bit and now it can import the biomass function from the .csv file

dmccloskey reviewed (changes requested):
Looking very nice.  Please let me know about the comments you had mentioned as I do not see them.  The only small issue is a missing docstrings for the new `biomass_function` parameter.

mmattano committed:
fix: test if it still works

mmattano committed:
fix: biomass function can be imported from .csv file

mmattano committed:
tests: adjusted tests file and data (biomass from .csv)

mmattano committed:
fix: adjusted example notebook

mmattano committed:
fix: oops, whitespace

dmccloskey reviewed (commented):
Biomass update looks good.  Let me know when the other comments are taken care of.

mmattano committed:
feat: moved everything to a class

Also moved the script to a separate folder

mmattano committed:
Merge branch 'develop' into MM_INCA_tools

mmattano committed:
fix: updated test script

mmattano committed:
Merge branch 'MM_INCA_tools' of https://github.com/mmattano/AutoFlow-OmicsDataHandling into MM_INCA_tools

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Update test_INCA_script_generator.py

mmattano committed:
Merge branch 'develop' into MM_INCA_tools

mmattano committed:
fix: updated example notebook

mmattano committed:
Merge branch 'MM_INCA_tools' of https://github.com/mmattano/AutoFlow-OmicsDataHandling into MM_INCA_tools
